umpost/conversion_driver_esm1p5.py

```diff
@@ -0,0 +1,201 @@
+        if isinstance(fields_file_path, str)
+        else fields_file_path
+    )
+
+    nc_write_dir = Path(nc_write_dir) if isinstance(nc_write_dir, str) else nc_write_dir
+
+    fields_file_name = fields_file_path.name
+    nc_name = fields_file_name + ".nc"
+    nc_write_path = nc_write_dir / nc_name
+
+    return nc_write_path
+
+
+def find_matching_fields_files(dir_contents, fields_file_name_pattern):
+    """
+    Find files in list of paths with names matching fields_file_name_pattern.
+    Used to find ESM1.5 UM outputs in a simulation output directory.
+
+    Parameters
+    ----------
+    dir_contents : list of file paths, typically contents of a single directory.
+    fields_file_name_pattern : Regex pattern for matching fields file names.
+
+    Returns
+    -------
+    fields_file_paths : subset of dir_contents with names matching fields_file_name_pattern.
+    """
+
+    fields_file_paths = []
+    for filepath in dir_contents:
+        filepath = Path(filepath)
+
+        if re.match(fields_file_name_pattern, filepath.name):
+            fields_file_paths.append(filepath)
+
+    return fields_file_paths
+
+
+def convert_fields_file_list(fields_file_path_list, nc_write_dir):
+    """
+    Convert group of fields files to NetCDF, writing output in nc_write_dir.
+
+    Parameters
+    ----------
+    fields_file_list : list of paths to fields files for conversion.
+    nc_write_dir : directory to save NetCDF files into.
+
+    Returns
+    -------
+    None
+    """
+
+    for fields_file_path in fields_file_path_list:
+
+        fields_file_path = (
+            Path(fields_file_path)
+            if isinstance(fields_file_path, str)
+            else fields_file_path
+        )
+
+        nc_write_path = set_nc_write_path(fields_file_path, nc_write_dir)
+
+        print("Converting file " + fields_file_path.name)
+
+        try:
+            um2netcdf4.process(fields_file_path, nc_write_path, ARG_VALS)
+
+        except Exception as exc:
+            # Not ideal here - um2netcdf4 raises generic exception when missing coordinates
+            if exc.args[0] in ALLOWED_UM2NC_EXCEPTION_MESSAGES.values():
+                warnings.warn("Unable to convert file: " + fields_file_path.name)
+            else:
+                raise
+
+
+def convert_esm1p5_output_dir(current_output_dir):
+    """Driver function for converting ESM1.5 atmospheric outputs during a simulation."""
+
+    current_output_dir = (
+        Path(current_output_dir)
+        if isinstance(current_output_dir, str)
+        else current_output_dir
+    )
+
+    current_atm_output_dir = current_output_dir / "atmosphere"
+
+    if not (current_atm_output_dir.exists()):
+        raise FileNotFoundError(
+            errno.ENOENT, os.strerror(errno.ENOENT), current_atm_output_dir
+        )
+
+    # Create a directory for writing NetCDF files
+    current_run_nc_dir = current_atm_output_dir / "NetCDF"
+    current_run_nc_dir.mkdir(exist_ok=True)
```

Comment:
Another stylistic idea from one of the clean code books. The var name `current_run_nc_dir` could be renamed to describe its intent as the nc output dir, then the code gains a self documenting quality & less comments are required.

Reply:
I've renamed it to `nc_write_dir` to try and make this clearer